### ISSUE TITLE
Handling of invalid XML characters in the Boost UTF reports/logs

### ISSUE THREAD

**Issue by gastart** [etas/vs-boost-unit-test-adapter#27]
The XML logs Boost UTF generates and that then BUTA consumes are not really well formatted because sometimes illegal characters lurk in. This causes BUTA to raise an exception and the user may be left wondering what went wrong.  

Even though it is an issue with Boost, ideally we do apply a workaround so that we clean up the xml file before consuming it in BUTA.

A typical file causing such issues is available at http://www.filedropper.com/testboostunittestexetestlog

```xml
<TestLog><TestSuite name="Test01"><TestSuite name="UnitTests"><TestCase name="UnitTest01"><Error file="d:/test/CTest.cpp" line="70"><![CDATA[check tester.Foo(0x20, 0x3F, byteBuffer) == 0 failed [2 != 0]]]></Error><Error file="d:/test/CTest.cpp" line="71"><![CDATA[check { byteBuffer.cbegin(), byteBuffer.cend() } == { byteBuffer.cbegin(),byteBuffer.cend() } failed. 
Mismatch in a position 0: H !=  
Mismatch in a position 2:  !=  
Mismatch in a position 3: ¦ !=  
Mismatch in a position 8: H !=  
Mismatch in a position 9:  !=  
Mismatch in a position 11: * !=  
Mismatch in a position 16: H !=  
Mismatch in a position 18:  !=  
Mismatch in a position 19: Æ !=  
Mismatch in a position 24: H !=  
Mismatch in a position 26:  !=  
Mismatch in a position 27: æ !=  
Collections size mismatch: 32 != 255]]></Error><TestingTime>9414000</TestingTime></TestCase></TestSuite></TestSuite></TestLog>
```

**Comment by guwirth** [comment]
Not really familiar with the .NET XML reader but normally character checking can be turned off (https://msdn.microsoft.com/en-us/library/system.xml.xmlreadersettings.checkcharacters(v=vs.95).aspx) or from Xerces I know that they have an option to replace it with a default character. Maybe same thing is possible here.

Anyway wondering how there could be invalid characters in the file? Is this not more an issue that the wrong encoding is used to read the file?

**Comment by gastart** [comment]
The problem seems to be the null termination character and since it is a control character AFAIK it is disallowed in the XML file. We were actually exploring the option of the character replacement as you also hinted to.

**Comment by guwirth** [comment]
Fixed with #31
